### PR TITLE
RavenDB-11746 Failed to properly close RavenServer (Delegate to an in…

### PR DESCRIPTION
**Pull request by ayende**
…stance method cannot have null 'this'.) when server configuration is broken

### CODE REVIEW

File: src/Raven.Server/Program.cs [hunk 1]
```diff
@@ -202,15 +202,23 @@ public static int Main(string[] args)
...
+                            {
                                 Logger.Operations("Failed to initialize the server", e);
+                                Logger.Operations(message);
+                            }
+
                             Console.WriteLine(message);
 
+                            Console.WriteLine();
+
+                            Console.WriteLine(e);
```
Comment (ppekrol) [inline src/Raven.Server/Program.cs:220]: This line reverts assumption from http://issues.hibernatingrhinos.com/issue/RavenDB-11107

Comment (ppekrol) [reply src/Raven.Server/Program.cs:220]: maybe wrap this in `#if DEBUG` ?

Comment (ayende) [reply src/Raven.Server/Program.cs:220]: We need this to get a clear error for _us_ in prod.
When a user runs into a problem, we need to have the full details.
You can see that the message is still shown first, along with the full error stack that we can use.

Comment (ayende) [reply src/Raven.Server/Program.cs:220]: Right now, see what happens when you configure a bad certificate and start the server.

**Pull request merged**